### ISSUE TITLE
Suggestion: Don't mute remote speech when controlling remote machine

### ISSUE THREAD

**Issue by Simon818** [NVDARemote/NVDARemote#282]
I found myself disconnecting from remote machines pretty frequently, and realized I was doing it because I didn't want messages coming through while I wasn't controlling it. So I patched my copy so that the mute toggle only applies when I'm controlling the local machine, and I find this to be significantly better than the original mute command. I can't really think of a valid use case for wanting to control the remote machine without hearing speech, but if we don't mind one more translated string it could be added as a checkbox in options for the people who still want the old behavior. Personally, I prefer it this way, but maybe the permanent mute is deliberate.

**Comment by ChrisDuffley** [comment]
I agree. Good option but not something I'm probably going to enable, simply because I do prefer hearing it even when controlling the local machine. I can definitely see this useful for some, though.

Christopher Duffley

Host of Mission Possible Podcast <http://www.missionpossiblepodcast.com/>

LINKS TO SUBSCRIBE:

iTunes <https://itunes.apple.com/us/podcast/mission-possible-christopher/id1103092109?mt=2> | Stitcher <http://www.stitcher.com/s?fid=88094&refid=stpr> | Google Play <https://goo.gl/app/playmusic?ibi=com.google.PlayMusic&isi=691797987&ius=googleplaymusic&link=https://play.google.com/music/m/Ik76dexklcprwnpuxbdr2s6bm3y?t=Mission_Possible_with_Christopher_Duffley> | Follow Podcast On TuneIn And Listen To Latest Episode <http://tunein.com/radio/Mission-Possible-With-Christopher-Duffley-p892505/>

LIKE/FOLLOW Me:

Facebook <http://facebook.com/christopherduffley> | Twitter <http://twitter.com/chrisduffley> | YouTube <http://youtube.com/christopherduffley> | Instagram <http://instagram.com/christopherduffley>

On Oct 9, 2021, at 14:51, Simon Jaeger ***@***.***> wrote:


I found myself disconnecting from remote machines pretty frequently, and realized I was doing it because I didn't want messages coming through while I wasn't controlling it. So I patched my copy so that the mute toggle only applies when I'm controlling the local machine, and I find this to be significantly better than the original mute command. I can't really think of a valid use case for wanting to control the remote machine without hearing speech, but if we don't mind one more translated string it could be added as a checkbox in options for the people who still want the old behavior. Personally, I prefer it this way, but maybe the permanent mute is deliberate.

**Comment by lbk2907** [comment]
How about if NVDARemote user have an option to tougle this feature on or
off. It will be grate.

**Comment by cary-rowen** [comment]
Hi @tech10 

I believe you can easily achieve this.
You can simply press NVDA+Ctrl+S on the remote computer and select "No speech".

Hope this works for you

**Comment by tech10** [comment]
This wouldn't exactly work to implement the suggestion stated here, as speech would still be disabled on the remote system if the addon disconnected for any reason. The goal of this suggestion, I believe, was to mute speech from the remote computer if it wasn't being controlled, presented as an option for people to enable or disable. Here's an example.

Computer A is controlling computer B, and receiving speech from computer B when something occurs. Computer A has set speech to be muted from computer B, which will only occur when computer A is not controlling computer B, and instead, controlling its own system. When computer A is controlling itself, and something occurs on computer B, no speech will be presented to computer A, but when computer A is controlling computer B, speech feedback is presented to computer A from computer B.

I believe this could be achieved by creating a mute dialog with some radio buttons, mute speech, or, mute speech when remote computer is not being controlled. And an unmute speech option to disable this feature. Of course, the addon would have to be modified appropriately to make this particular part of the interface work correctly.

**Comment by cary-rowen** [comment]
I believe I understand what you're saying, the approach I've mentioned really doesn't fully cover this suggested case.
This feature is necessary, is anyone willing to implement it?

**Comment by ctoth** [comment]
I really don't know that it is necessary, and it will add quite a bit of complication, translation burden and documentation for something that few people will use and that can be used to scam unaware users. 

Consider the case of a malicious attacker who connects to a user and sets their NVDA to not speak when being controlled so they don't even know they're compromised. 

We could preserve the synthesizer in use when the session starts and restore that upon session ending, but this would preclude support scenarios like someone helping another user change their synthesizer intentionally. 

In all I don't see much use for this feature and suspect that resources could best be spent elsewhere.

**Comment by beqabeqa473** [comment]
I think case is contrary, i mean to not receive speech from remote end
when we are not controlling it, so if we press f11, and we are back to
local machine, make it possible to not interrupt our own speech by
remote end, and restore it when we are back to controll.
It is the same as muting remote speech, but to be automatic when we
are leaving control if i understood corectly the case.

On 2/10/22, Christopher Toth ***@***.***> wrote:
> I really don't know that it is necessary, and it will add quite a bit of
> complication, translation burden and documentation for something that few
> people will use and that can be used to scam unaware users.
>
> Consider the case of a malicious attacker who connects to a user and sets
> their NVDA to not speak when being controlled so they don't even know
> they're compromised.
>
> We could preserve the synthesizer in use when the session starts and restore
> that upon session ending, but this would preclude support scenarios like
> someone helping another user change their synthesizer intentionally.
>
> In all I don't see much use for this feature and suspect that resources
> could best be spent elsewhere.
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/NVDARemote/NVDARemote/issues/282#issuecomment-1034572924
> You are receiving this because you are subscribed to this thread.
>
> Message ID: ***@***.***>


-- 
with best regards Beqa Gozalishvili
Tell: +995593454005
Email: ***@***.***
Web: https://gozaltech.org
Skype: beqabeqa473
Telegram: https://t.me/gozaltech
facebook: https://facebook.com/gozaltech
twitter: https://twitter.com/beqabeqa473
Instagram: https://instagram.com/beqa.gozalishvili

**Comment by beqabeqa473** [comment]
As for turning off speech at remote end itself, yes, i don't agree
with that also, it opens a security risk, and i don't think it is good
to go this way.

On 2/10/22, Beqa Gozalishvili ***@***.***> wrote:
> I think case is contrary, i mean to not receive speech from remote end
> when we are not controlling it, so if we press f11, and we are back to
> local machine, make it possible to not interrupt our own speech by
> remote end, and restore it when we are back to controll.
> It is the same as muting remote speech, but to be automatic when we
> are leaving control if i understood corectly the case.
>
> On 2/10/22, Christopher Toth ***@***.***> wrote:
>> I really don't know that it is necessary, and it will add quite a bit of
>> complication, translation burden and documentation for something that few
>> people will use and that can be used to scam unaware users.
>>
>> Consider the case of a malicious attacker who connects to a user and sets
>> their NVDA to not speak when being controlled so they don't even know
>> they're compromised.
>>
>> We could preserve the synthesizer in use when the session starts and
>> restore
>> that upon session ending, but this would preclude support scenarios like
>> someone helping another user change their synthesizer intentionally.
>>
>> In all I don't see much use for this feature and suspect that resources
>> could best be spent elsewhere.
>>
>> --
>> Reply to this email directly or view it on GitHub:
>> https://github.com/NVDARemote/NVDARemote/issues/282#issuecomment-1034572924
>> You are receiving this because you are subscribed to this thread.
>>
>> Message ID: ***@***.***>
>
>
> --
> with best regards Beqa Gozalishvili
> Tell: +995593454005
> Email: ***@***.***
> Web: https://gozaltech.org
> Skype: beqabeqa473
> Telegram: https://t.me/gozaltech
> facebook: https://facebook.com/gozaltech
> twitter: https://twitter.com/beqabeqa473
> Instagram: https://instagram.com/beqa.gozalishvili
>


-- 
with best regards Beqa Gozalishvili
Tell: +995593454005
Email: ***@***.***
Web: https://gozaltech.org
Skype: beqabeqa473
Telegram: https://t.me/gozaltech
facebook: https://facebook.com/gozaltech
twitter: https://twitter.com/beqabeqa473
Instagram: https://instagram.com/beqa.gozalishvili

**Comment by ChrisDuffley** [comment]
@ctoth & @tspivey, I'm using NVDARemote 2.5 and I do not see anything about pausing speech. Is this actually in the release? I'd prefer it off, but I thought it would be in Options or the Remote sub menu but don't see it anywhere.

**Comment by tspivey** [comment]
It's not.

**Comment by wmhn1872265132** [comment]
I am looking forward to the realization of this function. I often need to remotely control other computers, but when I am not controlling other computers, I don’t want to hear the voice output of other computers. Frequently switching remote mute is still a bit troublesome. In addition, when I need When controlling other computers, I often forget to restore the voice output of the remote computer, resulting in me not being able to hear the other computer's voice.

**Comment by Simon818** [comment]
I think there was some confusion because of the way I worded the original comment. The most recent commenter has it right.

If I use my laptop to control my desktop, and I use the "mute remote" option in the laptop's NVDA>Tools>Remote menu, and then I switch back to controlling the desktop (which is the remote machine), currently I will hear no remote speech on the laptop because the remote is muted. I believe there are very few scenarios where this is wanted, and there is no warning to the user who may have forgotten they muted the remote. I am proposing that even when remote is muted, the laptop should still play speech and sounds when controlling the remote machine. This becomes the new default behaviour. A toggle is added to the options screen that says "Preserve muted state when controlling remote machine".

This will avoid the confusion of trying to control a machine but not hearing any speech from it, and it will also make it possible to stay connected for long periods of time but not hear sounds or speech from the other machine unless one is also actively using it.

Hope that helps.
